src/main/java/org/abhinavgpt/commenz/controllers/ReviewController.java

```diff
@@ -1,35 +1,51 @@
 package org.abhinavgpt.commenz.controllers;
 
+import org.abhinavgpt.commenz.services.reviews.ReviewService;
 import org.abhinavgpt.commenz.services.summary.SummaryService;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.List;
+
 @RestController
 @RequestMapping("/api/review")
-public class ReviewController {
-
-    private final SummaryService summaryService ;
-
-    public ReviewController(SummaryService summaryService) {
-        this.summaryService = summaryService;
-    }
+public class ReviewController
+{
+	private final SummaryService summaryService;
+	private final ReviewService reviewService;
 
-    @GetMapping("")
-    public ResponseEntity<String> getReviews() {
-        try {
-            return ResponseEntity.ok(summaryService.greetingMessage());
-        } catch (Exception e) {
-            return ResponseEntity.badRequest().body(e.getMessage());
-        }
-    }
+	// We don't need this method - do we?
+//	@GetMapping("")
+//	public ResponseEntity<String> getReviews() // What should this do?
+//	{
+//		try
+//		{
+//			return ResponseEntity.ok(summaryService.greetingMessage());
+//		} catch (Exception e)
+//		{
+//			return ResponseEntity.badRequest().body(e.getMessage());
+//		}
+//	}
 
-    @PostMapping("/summarize")
-    public ResponseEntity<String> summarize(@RequestBody String url) {
-        try {
-            return ResponseEntity.ok(summaryService.getSummary(url));
-        } catch (Exception e) {
-            return ResponseEntity.badRequest().body(e.getMessage());
-        }
-    }
+	// changed this to Get from Post and @RequestParam from @RequestBody
+	@GetMapping("/summarize")
+	public ResponseEntity<String> summarize(@RequestParam String url)
+	{
+		List<String> reviews = reviewService.getReviews(url);
```

Comment:
would you call this as business logic?

Reply:
Controllers should be dumb. They just take the request and pass it to the service and now whatever needs to be done in order to fulfill that request it should be done by the service.

Reply:
Yes, that's what I saying before, that we need to make another service that will call the reviews service, get reviews and then call the summary service, then return it.

Reply:
ok then go ahead and implement it!